### PR TITLE
Feature/141 - simplify dataset adapter

### PR DESCRIPTION
**Pull request by sumitpai**

### CODE REVIEW

File: ampligraph/datasets/abstract_dataset_adapter.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def set_filter(self, filter_triples):
         """
         raise NotImplementedError('Abstract Method not implemented!')
         
-    def get_next_batch(self, batches_count=-1, dataset_type="train"):
+    def get_next_batch(self, batches_count=-1, dataset_type="train", use_filter=False):
```
Comment (NicholasMcCarthy) [inline ampligraph/datasets/abstract_dataset_adapter.py]: Was going to suggest this change but you beat me to it :)

Comment (sumitpai) [reply ampligraph/datasets/abstract_dataset_adapter.py]: I think you did mention it on the issue.

File: ampligraph/datasets/abstract_dataset_adapter.py [hunk 2]
```diff
@@ -98,30 +98,17 @@ def get_next_batch(self, batches_count=-1, dataset_type="train"):
             indicates which dataset to use
         batches_count: int
             number of batches per epoch (default: -1, i.e. uses batch_size of 1)
```
Comment (NicholasMcCarthy) [inline ampligraph/datasets/abstract_dataset_adapter.py]: The batches_count=-1 -> batch_size=1 is a little awkward but I don't see a better way of implementing it other than switching entirely to batch_size.

Comment (NicholasMcCarthy) [reply ampligraph/datasets/abstract_dataset_adapter.py]: (Which I'm not suggesting we do.)

**Pull request merged**